pkg/flow/flow_services.go

```diff
@@ -52,3 +55,38 @@ func serviceConsumersForGraph(graph *dag.Graph, serviceName string, includePeerS
 
 	return consumers
 }
+
+// NewController returns a new, unstarted, isolated Flow controller so that
+// services can instantiate their own components.
+func (f *Flow) NewController(id string) service.Controller {
+	return serviceController{
+		f: newController(controllerOptions{
+			Options: Options{
+				ControllerID:    id,
+				Logger:          f.opts.Logger,
+				Tracer:          f.opts.Tracer,
+				DataPath:        f.opts.DataPath,
+				Reg:             f.opts.Reg,
+				Services:        f.opts.Services,
+				OnExportsChange: nil, // NOTE(@tpaschalis, @wildum) The isolated controller shouldn't be able to export any values.
+			},
+			IsModule:       true,
```

Comment:
cc @rfratto as @mattdurham pointed out, we likely _don't_ want this new controller to run as a module controller, so that remote configuration can manage its own logging/tracing block for example.

Any concerns in regards to that?

Reply:
Eh, I see. Then I think it's something we can work out later on (and after modules v2.0 are out).

Reply:
Yeah these isModule conditions spread around the controller are scary. This is something that we should be able to address in the refactor of the controller

Reply:
How do we enforce what they can or cannot run in the main config when using remote config? We dont necessarily need it for this pr but I think it needs to be addressed before we productionize it.

Reply:
Sorry for jumping the gun here, I should have made sure we'd finalized the discussion first.

Since we mark the new controller with `IsModule: true`, then controller code will inherit the same services as run by the root controller. As such, if the remotecfg controller detects a config block, it will reject the new config being applied.